[JIT] Fix AliasDB edge case for multi non existing alias sets in output

ttanpcs opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82029
* #81917
* #81916
* __->__ #81837
* #81836
* #81786
* #81785

- Although currently no ops have this form, it's possible to have an op of the form: op(Tensor a, Tensor b) -> Tensor(c | d)
Since the alias sets c and d both don't exist in the input arguments, the output should call createNewAlias(), however previously, this wasn't the case because it assumes that if there are two alias sets on an output, at least one will also exist in the input arguments which isn't necessarily the case. 
- This is tested through all AliasDB tests passing in CI.

Commit:
[JIT] Fix AliasDB edge case for multi non existing alias sets in output

[ghstack-poisoned]

davidberard98 reviewed (commented):
nice! can you also add a test for the previously-broken situation?

davidberard98 reviewed (approved):
discussed offline, it's kinda hard to test this so I'll accept - if you have time it would be great if you can try to add a test using a custom op, but for now this change is fine :+1:

https://github.com/pytorch/pytorch/blob/master/test/cpp/jit/test_custom_class_registrations.cpp#L401

Commit:
Update on "[JIT] Fix AliasDB edge case for multi non existing alias sets in output"


- Although currently no ops have this form, it's possible to have an op of the form: op(Tensor a, Tensor b) -> Tensor(c | d)
Since the alias sets c and d both don't exist in the input arguments, the output should call createNewAlias(), however previously, this wasn't the case because it assumes that if there are two alias sets on an output, at least one will also exist in the input arguments which isn't necessarily the case. 
- This is tested through all AliasDB tests passing in CI.



[ghstack-poisoned]

Commit:
Update on "[JIT] Fix AliasDB edge case for multi non existing alias sets in output"


- Although currently no ops have this form, it's possible to have an op of the form: op(Tensor a, Tensor b) -> Tensor(c | d)
Since the alias sets c and d both don't exist in the input arguments, the output should call createNewAlias(), however previously, this wasn't the case because it assumes that if there are two alias sets on an output, at least one will also exist in the input arguments which isn't necessarily the case. 
- This is tested through all AliasDB tests passing in CI.



[ghstack-poisoned]

osalpekar commented:
@pytorchbot revert -m "Need to revert this to revert https://github.com/pytorch/pytorch/pull/81836 cleanly. That PR broke internal mobile custom_ops" -c ghfirst
